pyproject.toml

```diff
@@ -1,8 +1,88 @@
+[project]
```

Comment:
Side-note: are all the entries still present here?

Reply:
Not sure what you mean by this ...

Some of the metadata in `pyproject.toml` were required for building the wheels. I tried to move the `extra_requires` entries (`optional_dependencies` in `pyproject.toml`) although I am not sure they are used too much e.g. are people doing `pip install -e .[docs]`, or `pip install sklearn[tests]`, I certainly don't but others might ... 

If you see other things that need to be added, let me know!

Reply:
I like having a simple command to install all the build doc dependencies or the tests dependencies.

Ideally I would like conda/mamba to be able to pick them up as well but I don't know if it's possible. Anyways this is already an improvement.

Reply:
I dug a bit further and there are ongoing discussions but no definite solution at this point:

- https://github.com/conda/conda/issues/10633
- https://conda.github.io/conda-lock/src_pyproject/#lock-only-conda-lock-dependencies
- https://peps.python.org/pep-0735/